File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int sz = nums.size(), maxi = 0, cnt = 0;

        for(int i=0; i<sz; i++){
            int j=i, tmp = 0;
            while(j < sz && nums[i] == nums[j]){
                tmp++; j++;
            }
            if(maxi < nums[i]){
                maxi = nums[i];
                cnt = tmp;       
            }
            else if(maxi == nums[i]) cnt = max(cnt, tmp);
            j--;
            i = j;
        }

        return cnt;
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int sz = nums.length, cnt = 0, maxi = -1, ans = 0;

        for(int i : nums) maxi = maxi > i ? maxi : i;

        for(int i=0; i<sz; i++){
            cnt = 0;
            while(i < sz && nums[i] == maxi){
                cnt++;
                i++;
            }
            ans = ans > cnt ? ans : cnt;
        }

        return ans;
    }
}